onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1923,6 +1924,349 @@ Matrix product that behaves like numpy.matmul: https://docs.scipy.org/doc/numpy-
         FusedMatMulShapeInference(ctx);
       });
 
+  static const char* NonZero_doc = R"DOC(
+  Behaves similar to https://github.com/onnx/onnx/blob/master/docs/Operators.md#NonZero
+  returns a tensor that contains coordinates for non-zero entries. Even though the input
+  is a sparse tensor, that usually does not contain zeros, the model logic may construct
+  sparse tensor containing explicit zeros that convey model specific meaning.
+
+  The output dense tensor would contain in a row-major order, by dimension, indices of non-zero elements
+)DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(NonZero)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "A", "2-dimensional sparse matrix A. Either COO format either 1-D or 2-D index", "T")
+      .Output(0, "B", "1 or 2-dimensional dense matrix B", "T1")
+      .TypeConstraint("T", {"sparse_tensor(float)"}, "Constrain to float at this time")
+      .TypeConstraint("T1", {"tensor(int64)"}, "Index type")
+      .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        // Output type is index type
+        propagateElemTypeFromDtypeToOutput(ctx, TensorProto_DataType_INT64, 0);
+        TensorShapeProto output_shape;
+        auto* dim = output_shape.add_dim();
+        if (hasInputShape(ctx, 0)) {
+          const TensorShapeProto& input_shape = getInputShape(ctx, 0);
+          dim->set_dim_value(input_shape.dim_size());
+        }
+        output_shape.add_dim();
+        updateOutputShape(ctx, 0, output_shape);
+      });
+
+  static const char* SparseGather_doc = R"DOC(
+This is a simplified version of ONNX Gather. No axis support.
+The input 0 is a sparse tensor in COO format
+The input 1 is a dense tensor containing COO 1-D indices of elements to gather.
```

Comment:
We do not support negative indices, although we support negative axis for some ops. Since Gather is a standard op, we want to make it consistent with the current spec or make minimal changes.